virtcontainers/qemu.go

```diff
@@ -609,6 +626,35 @@ func (q *qemu) createSandbox(ctx context.Context, id string, networkNS NetworkNa
 	return nil
 }
 
+func (q *qemu) setVSockHostPort() error {
+	socket, err := unix.Socket(unix.AF_VSOCK, unix.SOCK_STREAM, 0)
+	if err != nil {
+		return err
+	}
+	defer syscall.Close(socket)
```

Comment:
There is a race condition here: we fetch the port number and then close the file descriptor.  Now another host process can reuse the same port number before we get a chance to call vsock.Listen().  When this happens our vsock.Listen() call will fail.

Please keep the file descriptor open and use it instead of calling vsock.Listen() later on.

Reply:
okay, thanks. I'll save the server socket somewhere for later use.